The following scripts enable identification and calculation of an enrichment score for expression of co-regulated interactomes in transcriptional data.

In the first script, we identify co-regulated gene networks used as transcriptional modules. This requires a file of predicted upstream regulators associated with their (comma separated) target gene list, and the genome-wide expression matrix from an experimental data set. We calculate the average correlation coefficient for pairwise correlations of the expression levels of each group of target genes associated with predicted upstream regulators in our transcriptomic data set, and compared this to the distribution of average correlation coefficients obtained from random selection of equivalent sized groups of genes repeated 100 times. Groups of target genes with average correlation coefficients that exceeded the mean of the distribution of equivalent sized randomly selected groups by ≥2 SD (z-score ≥2) with false discovery rate <0.05 are selected as transcriptional modules representing the functional activity of the associated upstream regulator.

In the second script, we transform average gene expression values for each of the selected upstream regulator modules to standardised (Z scores) using mean and standard deviation of 100 randomly selected gene sets of the same size within each individual data set.
